Dockerfile

```diff
@@ -31,10 +31,14 @@ ENV GOFLAGS="-mod=readonly"
 RUN go build ./...
 
 ARG VERSION
-ENV GOFLAGS="-mod=readonly -ldflags=-X=github.com/improbable-eng/etcd-cluster-operator/version.Version=${VERSION}"
+ARG RESTORE_AGENT_IMAGE
 
 # Compile all the binaries
-RUN go build -o manager main.go
+RUN go build \
+    -mod=readonly \
+    -ldflags="-X=github.com/improbable-eng/etcd-cluster-operator/version.Version=${VERSION} -X=main.defaultRestoreAgentImage=${RESTORE_AGENT_IMAGE}" \
+    -o manager main.go
```

Comment:
Make the default restore-agent image match the latest built docker image name means we don't have to bother supplying the restore-agent image flag to the controller. It'll just use the docker image that was built at the same time as it.